openedx/adg/lms/applications/views.py

```diff
@@ -135,3 +138,96 @@ def get_context_data(self, **kwargs):
         context = super(ApplicationSuccessView, self).get_context_data(**kwargs)
         context['submission_date'] = self.request.user.application_hub.submission_date
         return context
+
+
+class CoverLetterView(RedirectToLoginOrRelevantPageMixin, View):
+    """
+    View enabling the user to select a Business Line and upload or write a cover letter.
+    """
+
+    template_name = 'adg/lms/applications/cover_letter.html'
+
+    def is_precondition_satisfied(self):
+        """
+        Checks if a written application is already submitted or not.
+
+        Returns:
+            Boolean, True or False.
+        """
+        user_application_hub, _ = ApplicationHub.objects.get_or_create(user=self.request.user)
+
+        return not user_application_hub.is_written_application_completed
+
+    def handle_no_permission(self):
+        """
+        Redirects to application hub on get request or returns htt 400 on post request.
+
+        Returns:
+            HttpResponse object.
+        """
+        if self.request.method == 'POST':
+            return HttpResponse(status=400)
+        else:
+            return redirect('application_hub')
+
+    def get(self, request):
+        """
+        Send the context data for example a list of business lines and saved user_application.
+
+        Returns:
+            HttpResponse object.
+        """
+        business_lines = BusinessLine.objects.all()
+        file_name = None
+
+        try:
+            user_application = UserApplication.objects.get(user=request.user)
+            file_name = Path(user_application.cover_letter_file.name).name
+        except UserApplication.DoesNotExist:
+            user_application = None
+
+        context = {
+            'business_lines': business_lines,
+            'logo_url': 'http://localhost:18000/media/',
+            'user_application': user_application,
+            'csrf_token': get_token(request),
+            'filename': file_name
+        }
+
+        return render(request, self.template_name, context)
+
+    def post(self, request):
+        """
+        Submit user application.
+
+        Returns:
+            HttpResponse object.
+        """
+        user_application, _ = UserApplication.objects.get_or_create(user=request.user)
+
+        if 'business_line' in request.POST and request.POST['business_line']:
```

Comment:
Do we need these checks when we are already doing front end validations, and making sure that business line is selected. A post request won't be triggered unless a business line is selected by the user. I think you wrote that logic in cover_letter.js `showpop()` function

Reply:
Just a side note, if we need to do backend validations, then it is preferred to override the `clean()` method in the form.